src/main/java/org/folio/service/protection/ProtectionServiceImpl.java

```diff
@@ -153,23 +166,28 @@ private List<String> getProvidedPermissions(Map<String, String> headers) {
    * Verifies if all acquisition units exist and active based on passed ids
    *
    * @param acqUnitIds list of unit IDs.
-   * @return completable future completed successfully if all units exist and active or exceptionally otherwise
+   * @return future completed successfully if all units exist and active or exceptionally otherwise
    */
-  public CompletableFuture<Void> verifyIfUnitsAreActive(List<String> acqUnitIds, String lang, Context context, Map<String, String> headers) {
+  public Future<Void> verifyIfUnitsAreActive(List<String> acqUnitIds, String lang, Context context, Map<String, String> headers) {
     logger.debug("verifyIfUnitsAreActive:: Trying to verify if units are active by acqUnitsIds: {}", acqUnitIds);
+    Promise<Void> promise = Promise.promise();
     if (acqUnitIds.isEmpty()) {
-      return CompletableFuture.completedFuture(null);
+      promise.complete();
+      return promise.future();
     }
-
-    return getUnitsByIds(acqUnitIds, lang, context, headers).thenAccept(units -> {
+    return getUnitsByIds(acqUnitIds, lang, context, headers).compose(units -> {
       List<String> activeUnitIds = units.stream()
         .filter(unit -> !unit.getIsDeleted())
         .map(AcquisitionsUnit::getId)
         .collect(Collectors.toList());
 
       if (acqUnitIds.size() != activeUnitIds.size()) {
-        throw new HttpException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY.toInt(), buildUnitsNotFoundError(acqUnitIds, activeUnitIds));
+        promise.fail(new HttpException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY.toInt(), buildUnitsNotFoundError(acqUnitIds, activeUnitIds)));
```

Comment:
You need to add return here, because next line can ovveride your fail promise

Reply:
I think we can replace this method something like this:
`  public Future<Void> verifyIfUnitsAreActive(List<String> acqUnitIds, String lang, Context context, Map<String, String> headers) {
    logger.debug("verifyIfUnitsAreActive:: Trying to verify if units are active by acqUnitsIds: {}", acqUnitIds);
    if (acqUnitIds.isEmpty()) {
      return Future.succeededFuture();
    }
    return getUnitsByIds(acqUnitIds, lang, context, headers).compose(units -> {
      List<String> activeUnitIds = units.stream()
        .filter(unit -> !unit.getIsDeleted())
        .map(AcquisitionsUnit::getId)
        .collect(Collectors.toList());

      if (acqUnitIds.size() != activeUnitIds.size()) {
        return Future.failedFuture(new HttpException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY.toInt(), buildUnitsNotFoundError(acqUnitIds, activeUnitIds)));
      }

      return Future.succeededFuture();
    });
  }`

Reply:
fixed